hikari/impl/event_manager_base.py

```diff
@@ -448,9 +451,10 @@ def get_listeners(
     ) -> typing.Collection[event_manager_.CallbackT[event_manager_.EventT_co]]:
         if polymorphic:
             listeners: typing.List[event_manager_.CallbackT[event_manager_.EventT_co]] = []
-            for subscribed_event_type, subscribed_listeners in self._listeners.items():
-                if issubclass(subscribed_event_type, event_type):
-                    listeners += subscribed_listeners
+            for event in event_type.dispatches():
+                if subscribed_listeners := self._listeners.get(event):
+                    listeners.extend(subscribed_listeners)
```

Comment:
This changes the nature of the method and breaks from the documented behaviour for polymorphic

Reply:
Updated docstring. Will add a `FIXME` to cause the pipelines to fail and make sure to add a breaking changelog fragment